From 1c0a2e1510b9d7b3dd6a637f4865451ecaad0b63 Mon Sep 17 00:00:00 2001
From: Mel Gorman <mgorman@suse.de>
Date: Mon, 18 May 2015 15:19:42 +0100
Subject: [PATCH] mm: fix huge zero page accounting in smaps report -s390

References: Automatic NUMA Balancing (fate#315482)
Patch-mainline: No, never

This is an s390-specific stub function to allow smaps to build properly. The
information is not critical so on its own does not justify a backport of
commit 152125b7a882df36a55a8eadbea6d0edf1461ee7.

Signed-off-by: Mel Gorman <mgorman@suse.de>
---
 arch/s390/include/asm/pgtable.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/s390/include/asm/pgtable.h b/arch/s390/include/asm/pgtable.h
index 886c38a0a7eb..cd1167717d08 100644
--- a/arch/s390/include/asm/pgtable.h
+++ b/arch/s390/include/asm/pgtable.h
@@ -605,6 +605,17 @@ static inline int pmd_write(pmd_t pmd)
 	return (pmd_val(pmd) & _SEGMENT_ENTRY_PROTECT) == 0;
 }
 
+/*
+ * Note that this is a stub only that is never expected to be used. At the
+ * time of writing, only task_mmu.c cared for smaps accounting and it's
+ * harmless to ignore it as an alternative to backporting
+ * 152125b7a882df36a55a8eadbea6d0edf1461ee7
+ */
+static inline int pmd_dirty(pmd_t pmd)
+{
+	return 0;
+}
+
 static inline int pmd_young(pmd_t pmd)
 {
 	int young = 0;
